app/src/main/java/com/nevilleantony/prototype/adapters/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,81 @@
+package com.nevilleantony.prototype.adapters;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.nevilleantony.prototype.R;
+import com.nevilleantony.prototype.downloadmanager.DownloadRepo;
+import com.nevilleantony.prototype.downloadmanager.FileDownload;
+import com.nevilleantony.prototype.utils.Utils;
+
+import java.util.ArrayList;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewHolder> {
+
+    Context ct;
+    DownloadRepo downloadRepo = DownloadRepo.getInstance(ct);
+    ArrayList downloads = downloadRepo.getDownloads();
+
+    public RecyclerViewAdapter(Context ct) {
+        this.ct = ct;
+    }
```

Comment:
Don't store context unless absolutely needed. This block of code should be changed to:

```java
    private final DownloadRepo downloadRepo;
    private List<FileDownload> downloads;

    public RecyclerViewAdapter(Context context) {
        downloads = new ArrayList<>();
        downloadRepo = DowloadRepo.getInstance(context);
        downloads.addAll(downloadRepo.getDownload());
    }
```